### PR TITLE
Extract IDebtToken from ILoanToken interface and use it where possible

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

**Comment by yuchenlintt** [comment]
Now that I read through this PR, I'm starting to think we may not want to inherit IDebtToken from ILoanToken2 at all.

Instead, I think we should break backward compatibility with LoanToken2 altogether and _just_ use a much simpler DebtToken (or name this LoanToken3?) to handle both LOC defaults and all FTL loans. We could remove all of the LoanToken2 logic related to:
- admin (only used to set the `transferable` boolean, which looks rather sketchy to me)
- transfer restrictions (irrelevant now since we're only allowing `liquidExit()` from pools)
- awaiting + funding + withdrawal (will be combined in a single step during creation)
- liquidation (does this actually do anything apart from mark the token? is it still relevant after we remove loan rating?)

There would only be three different statuses: {Withdrawn, Settled, Defaulted}. In the same tx an FTL is constructed (or its proxy initialized), TrueLender2* transfers the `principal` to the borrower. The DebtToken would record `principal`, `interest`, `start`, and `term` (`apy()` wouldn't need to be stored; it can be calculated from `interest` and `term`). Just like what we did with borrower eligibility, `status()` transitions automatically via a view function:
```
if balance + redeemed >= principal + interest:
    return Settled
if block.timestamp > start + term:
    return Defaulted
return Withdrawn
```

When we call default on an LOC, we could just create a DebtToken _without_ transferring funds and set the parameters to something like:
- `start = block.timestamp`
- `term = 0`
- `principal = TCA.borrowed(pool, borrower)`
- `interest = TCA.interest(pool, borrower)`

So its status() immediately returns Defaulted, and its principal and interest reflect the principal and interest of the defaulted LOC. We could even transfer mutex `locker()` from TCA to the DebtToken by calling `mutex.unlock()` and `mutex.lock(borrower, debtToken)`.

The interface would look something like this:
```
interface IDebtToken is IERC20 {
    enum Status {Withdrawn, Settled, Defaulted}

    function pool() external view returns (ITrueFiPool2);

    function token() external view returns (ERC20);

    function decimals() external view returns (uint8);

    function borrower() external view returns (address);

    function status() external view returns (Status);

    function principal() external view returns (uint256);

    function interest() external view returns (uint256);

    function start() external view returns (uint256);

    function term() external view returns (uint256);

    function apy() external view returns (uint256);

    function debt() external view returns (uint256);

    function redeem(uint256 _amount) external;

    function repay(address _sender, uint256 _amount) external;

    function repayInFull(address _sender) external;

    function reclaim() external;

    function version() external pure returns (uint8);
}
```

---

*A separate discussion: since we're making such large breaking changes to how FTL works, would it make sense to migrate FTLs to a new agency contract and slowly deprecate TrueLender2? For example, FTLs could be created by a `FixedTermLoanAgency.borrow(pool, amount, term)`, and LOC withdrawals could come from a `LineOfCreditAgency.borrow(pool, amount)`. When we launch, current borrowers would start out mutex locked to TrueLender2 until their existing loans settle. Afterward, they can only choose between FTLA and LOCA. When TrueLender2 no longer has any loans we can do an upgrade to remove it from all TFP2s.

**Comment by yuchenlintt** [comment]
If we decide to continue with the current approach, then this PR LGTM.

But next time please consider breaking this up into smaller pieces. I think a cleaner division would have been:
1. One PR to introduce IDebtToken and have ILoanToken2 inherit from IDebtToken. Doesn't touch any other files. We would be able to discuss which functions we want in IDebtToken and which to keep in ILoanToken2.
2. A separate PR to just find/replace ILoanToken2 with IDebtToken, where appropriate.
3. A third PR to add the deprecation comments on PoolExtensions.

It's nicer to separate simple refactors that touch a lot of files (e.g., renaming) into their own dedicated PRs. The ones that take some discussion and thought (how should we split up responsibilities between different interfaces?) should touch as few files as possible so we can keep discussion focused.

**Comment by vanruch** [comment]
@yuchenlintt Agree with the LoanToken simplification idea and I think that this PR would be a good starting point for that.

At the moment looks like we can use same contract for handling loans and defaulted LoCs. In the future it's possible we'll need to separate them (for example if we will ever decide to have openly tradeable loans) but this should be easily doable with inheritance.

On the other topic, we should start thinking about deprecating TrueLender to make it more similar to TCA but let's leave it for later

File: contracts/truefi2/TrueFiPool2.sol [hunk 1]
```diff
@@ -572,8 +572,10 @@ contract TrueFiPool2 is ITrueFiPool2, IPauseableContract, ERC20, UpgradeableClai
     /**
      * @dev Function called by SAFU when liquidation happens. It will transfer all tokens of this loan the SAFU
      */
-    function liquidate(ILoanToken2 loan) external override {
-        PoolExtensions._liquidate(safu, loan, lender);
+    function liquidate(IDebtToken loan) external override {
+        require(msg.sender == address(safu), "TrueFiPool: Should be called by SAFU");
+        // TODO handle flow for LoC DebtToken
```
Comment (barrutko) [inline contracts/truefi2/TrueFiPool2.sol:577]: Let's create an Asana task for that already.

**Review by barrutko — APPROVED** [review]
LGTM.

**Review by skalermo — APPROVED** [review]

File: contracts/truefi/TrueFiPool.sol [hunk 2]
```diff
@@ -15,7 +15,8 @@ import {IUniswapRouter} from "./interface/IUniswapRouter.sol";
 import {ABDKMath64x64} from "./Log.sol";
 import {ICrvPriceOracle} from "./interface/ICrvPriceOracle.sol";
 import {IPauseableContract} from "../common/interface/IPauseableContract.sol";
-import {ITrueFiPool2, ITrueFiPoolOracle, ITrueLender2, ILoanToken2, ISAFU} from "../truefi2/interface/ITrueFiPool2.sol";
+import {ITrueFiPool2, ITrueFiPoolOracle, ITrueLender2, ISAFU} from "../truefi2/interface/ITrueFiPool2.sol";
```
[the remark this answers is not in the document]
Comment (yuchenlintt) [reply contracts/truefi/TrueFiPool.sol:18]: Yeah, let's do that in another PR.

**Review by yuchenlintt — APPROVED** [review]

**Pull request merged**